frontend/packages/dev-console/src/actions/add-resources.tsx

```diff
@@ -13,84 +13,121 @@ import {
-  ImportOptions.HELMCHARTS,
-);
-
-export const uploadJarFile = createKebabAction(
-  // t('devconsole~Upload JAR file')
-  'devconsole~Upload JAR file',
-  <FileUploadIcon />,
-  ImportOptions.UPLOADJAR,
-  allCatalogImageResourceAccess,
-);
+type AddActionItem = { id: string; action: KebabAction };
+
+export const fromGit: AddActionItem = {
+  id: 'import-from-git',
+  action: createKebabAction(
+    // t('devconsole~From Git')
+    'devconsole~From Git',
+    <GitAltIcon />,
+    ImportOptions.GIT,
+    allImportResourceAccess,
+  ),
+};
+
+export const containerImage: AddActionItem = {
+  id: 'deploy-image',
+  action: createKebabAction(
+    // t('devconsole~Container Image')
+    'devconsole~Container Image',
+    <OsImageIcon />,
+    ImportOptions.CONTAINER,
+    allCatalogImageResourceAccess,
+  ),
+};
+
+export const fromCatalog: AddActionItem = {
+  id: 'dev-catalog',
+  action: createKebabAction(
+    // t('devconsole~From Catalog')
+    'devconsole~From Catalog',
+    <CatalogIcon />,
+    ImportOptions.CATALOG,
+  ),
+};
+
+export const fromDockerfile: AddActionItem = {
+  id: 'import-from-dockerfile',
+  action: createKebabAction(
+    // t('devconsole~From Dockerfile')
+    'devconsole~From Dockerfile',
+    <CubeIcon />,
+    ImportOptions.DOCKERFILE,
+    allImportResourceAccess,
+  ),
+};
+
+export const fromDevfile: AddActionItem = {
+  id: 'import-from-devfile',
+  action: createKebabAction(
+    // t('devconsole~From Devfile')
+    'devconsole~From Devfile',
+    <LayerGroupIcon />,
+    ImportOptions.DEVFILE,
+    allImportResourceAccess,
+  ),
+};
+
+export const fromDatabaseCatalog: AddActionItem = {
+  id: 'dev-catalog-databases',
+  action: createKebabAction(
+    // t('devconsole~Database')
+    'devconsole~Database',
+    <DatabaseIcon />,
+    ImportOptions.DATABASE,
+  ),
+};
+
+export const fromSamples: AddActionItem = {
+  id: 'import-from-samples',
+  action: createKebabAction(
+    // t('devconsole~Samples')
+    'devconsole~Samples',
+    <LaptopCodeIcon />,
+    ImportOptions.SAMPLES,
+  ),
+};
+
+export const fromOperatorBacked: AddActionItem = {
+  id: 'operator-backed',
+  action: createKebabAction(
+    // t('devconsole~Operator Backed')
+    'devconsole~Operator Backed',
+    <BoltIcon />,
+    ImportOptions.OPERATORBACKED,
+  ),
+};
+
+export const fromHelmCharts: AddActionItem = {
+  id: 'helm',
+  action: createKebabAction(
+    // t('devconsole~Helm Charts')
+    'devconsole~Helm Charts',
+    <img style={{ height: '1em', width: '1em' }} src={helmIcon} alt="Helm Charts Logo" />,
+    ImportOptions.HELMCHARTS,
+  ),
+};
+
+export const uploadJarFile: AddActionItem = {
+  id: 'upload-jar',
+  action: createKebabAction(
+    // t('devconsole~Upload JAR file')
+    'devconsole~Upload JAR file',
+    <FileUploadIcon />,
+    ImportOptions.UPLOADJAR,
+    allCatalogImageResourceAccess,
+  ),
+};
+
+const disabledAddActions = getDisabledAddActions();
```

Comment:
This will invoke once on first import and then not watch / ever get updated SERVER_FLAGS data, right? Wouldn't this need a browser refresh to be updated?

Reply:
Any update to console config rolls out a new deployment of console which updates the SERVER_FLAGS. That again requires a browser refresh to get the new flags. So I think this is not going to be an issue.

Reply:
Interesting, we have live updates of SERVER_FLAGS, no? I'm only concerned about the response-ability of the app. But if you feel this isn't a loss of features (uniquely), I'll let it go. Some things like the Terminal not live updating into the masthead feels anti-modern app. I'm just trying to state against more of "that" kinda features.

Reply:
All of our features that rely on the console config CR actually have the same issue. The server flags are set using environment variables of the pod by our backend here https://github.com/openshift/console/blob/master/pkg/server/server.go#L498. It just updates the `index.html` with all the flag values. So it cannot have live updates. AFAIK, we do not have a way to get dynamic `SERVER_FLAGS` updates.
@christianvogt correct me if I am wrong.

Reply:
`SERVER_FLAGS` never change. A change to the config will roll out a new deployment of console. 
We don't need to watch this.

This has been brought up before but the server flags approach is legacy and best to bring it up at the console arch meeting.

Reply:
My bad 👍  Thanks guys.